integration/test.sh

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+script_dir="$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)"
+
+# shellcheck disable=SC1091
+. "$script_dir/test_utils.sh"
+
+# Ensure the Docker image is up-to-date unless NO_BUILD=1
+if [ -z "$NO_BUILD" ]; then
+    "$script_dir"/build.sh
+fi
+
+start_lexical() {
+    local command='LX_HALT_AFTER_BOOT=1 _build/dev/package/lexical/bin/start_lexical.sh; exit $?'
+
+    if [[ $1 != "" ]]; then
+        command="$1 && $command"
+    fi
+
+    docker run -i lx bash -c "$command" 2>&1
+    return $?
+}
+
+run_test() {
```

Comment:
Should this go in `test_utils.sh`?

Reply:
As it's currently written, I don't think so, because it calls `start_lexical` directly. I originally had a more abstracted version of it in `test_utils.sh`, but ended up moving it here because there was just a lot of nested output/exit code capture going on and I found it hard to follow/debug.

My inclination is to leave it here until some of this logic is needed elsewhere, then pull it out.

Reply:
Sounds like a good plan 👍